var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -540,38 +541,26 @@ def supported_languages(self):
     compiler_version_regex = r"(?<!clang version)\s?([0-9.]+)"
     compiler_version_argument = ("-dumpfullversion", "-dumpversion")
 
-    @classmethod
-    def determine_version(cls, exe):
-        try:
-            output = spack.compiler.get_compiler_version_output(exe, "--version")
-        except Exception:
-            output = ""
-        # Apple's gcc is actually apple clang, so skip it.
-        if "Apple" in output:
-            return None
-
-        return super().determine_version(exe)
-
     @classmethod
     def filter_detected_exes(cls, prefix, exes_in_prefix):
-        result = []
-        for exe in exes_in_prefix:
-            # On systems like Ubuntu we might get multiple executables
-            # with the string "gcc" in them. See:
-            # https://helpmanual.io/packages/apt/gcc/
-            basename = os.path.basename(exe)
-            substring_to_be_filtered = [
-                "c99-gcc",
-                "c89-gcc",
-                "-nm",
-                "-ar",
-                "ranlib",
-                "clang",  # clang++ matches g++ -> clan[g++]
-            ]
-            if any(x in basename for x in substring_to_be_filtered):
-                continue
+        # On systems like Ubuntu we might get multiple executables with the string "gcc" in them.
+        # See: https://helpmanual.io/packages/apt/gcc/
+        # Also, clang++ matches g++ -> clan[g++]
+        reject = re.compile(r"c99-gcc|c89-gcc|-nm|-ar|ranlib|clang|mingw")
```

Comment:
mingw is not the only cross compiler.

I would suggest to use the old regexes, they work and they use `^gcc` so you don't need to filter stuff out later.

Reply:
Mostly done, with the exception of `x86_64-linux-gnu-` prefix that we detect in the package class, but not in the compiler class.

Reply:
As requested https://github.com/spack/spack/pull/45810/commits/d091c58f573527209b70da5a3f4a84a6840dfa07 uses the old regexes, and doesn't detect any compiler with `<triplet>-gcc` names